compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/SyntaxTree.java

```diff
@@ -64,6 +65,29 @@ public static SyntaxTree from(SyntaxTree oldTree, TextDocumentChange textDocumen
                 newTextDocument, oldTree.filePath(), false);
     }
 
+    public static SyntaxTree fromTopLevel(TextDocument textDocument) {
+        return fromGivenContext(textDocument, null, ParserRuleContext.TOP_LEVEL_NODE);
+    }
+
+    public static SyntaxTree fromStatement(TextDocument textDocument) {
+        return fromGivenContext(textDocument, null, ParserRuleContext.STATEMENT);
+    }
+
+    public static SyntaxTree fromStatements(TextDocument textDocument) {
+        return fromGivenContext(textDocument, null, ParserRuleContext.STATEMENTS);
+    }
+
+    public static SyntaxTree fromExpression(TextDocument textDocument) {
+        return fromGivenContext(textDocument, null, ParserRuleContext.EXPRESSION);
+    }
+
+    public static SyntaxTree fromGivenContext(TextDocument textDocument, String filePath, ParserRuleContext context) {
+        // TODO: Can get rid of overloaded methods once ParserRuleContext is exposed to outside.
```

Comment:
@rdulmina Once we have a proper place to expose parser internals, we can remove the loaded methods. For now, this is okay? WDYT?

Reply:
We can also rename `fromGivenContext` to `fromContext`. Or a better name suggestion?

Reply:
+1 but the names doesn't make sense to me

Reply:
Names updated as per offline discussion. Therefore resolving this.